### PR TITLE
[ROCm][Perf] Enable the AITER GDN decode fast path for flat qkvz layouts

### PR DESCRIPTION
**Pull request by mjkvaak-amd**
## Purpose

`_forward_core_rocm` currently gates the AITER GDN decode fast path on `self.gqa_interleaved_layout`, so only Qwen3-Next takes it and flat-layout models (e.g. Qwen3.5, Qwen3.8) fall through to the generic path. #42880 added that guard because the fused reshape+conv kernel only understood Qwen3-Next's packing and read the wrong columns from a flat tensor.

ROCm/aiter#3251 added a `qkvz_layout` parameter handling both. It selects the source index mapping only — outputs stay in canonical order, so the downstream `fused_rearrange_sigmoid_gated_delta_rule` is unchanged — and `"interleaved"` stays the default, so Qwen3-Next is bit-identical. This PR passes the layout through and drops the layout guard. Older AITER builds would raise `TypeError`, so `gdn_fused_conv_supports_qkvz_layout()` probes the signature once (mirroring
`fused_moe_supports_gate_mode()`) and flat layouts keep the guard there.

The fast path returns before the entire generic block, so four kernel launches per GDN layer per step become two: `core_attn_out.zero_()` and `z_out[:] = z` are pure removals, since the AITER kernel writes both buffers including the padding rows above `num_actual_tokens`.

Also adds `vllm/model_executor/layers/mamba/gdn` to the Kernels Mamba `source_file_dependencies`, which did not list it.

## Test Plan

```bash
pytest tests/kernels/mamba/test_gdn_rocm_layout_dispatch.py -q
```

New host-side tests (CPU, no GPU or AITER needed) covering flat dispatch with and without kernel support, Qwen3-Next non-regression on both, batch shapes that must stay generic, and the capability probe. Accuracy uses the `lm_eval` invocation from #42880; performance uses decode traces of `amd/Qwen3.8-2.4T-A95B-Quark-MXFP4` on 8x MI355X.

## Test Result

```bash
14 passed.
```

**Performance:** Serving benchmark on `amd/Qwen3.8-2.4T-A95B-Quark-MXFP4`, 8x MI355X, TP8, fixed input/output lengths (`--random-range-ratio 0`), one run per point, measured against a stock build of the same commit.

| ISL  | OSL  | Conc | Output tok/s              | Mean TPOT (ms)        |
| ---- | ---- | ---- | ------------------------- | --------------------- |
| 1024 | 1024 | 4    | 250.6 → 258.6 (+3.2%)     | 15.66 → 15.16 (-3.2%) |
| 1024 | 1024 | 16   | 674.0 → 688.3 (+2.1%)     | 22.95 → 22.45 (-2.1%) |
| 1024 | 1024 | 64   | 1,560.7 → 1,594.7 (+2.2%) | 38.82 → 37.94 (-2.3%) |

**Accuracy:** GSM8K, 5-shot, seed 4321, `batch_size=128`:
| Run                 | flexible-extract | strict-match    |
| ------------------- | ---------------- | --------------- |
| Stock (main)        | 0.9318 ± 0.0069  | 0.9333 ± 0.0069 |
| This PR             | 0.9348 ± 0.0068  | 0.9363 ± 0.0067 |

No accuracy regression.

**(Decode) trace:**

| Kernel | Before | After |
|---|---:|---:|
| `core_attn_out.zero_()` | 3933 / 16.3 ms | absent |
| `direct_copy_kernel` (`z_out[:] = z`) | 3920 / 17.0 ms | absent |
| `_causal_conv1d_update_kernel` | 3933 / 16.7 ms | absent |
| `fused_recurrent_gated_delta_rule_packed_decode_kernel` | 3933 / 23.7 ms | absent |
| `_reshape_causal_conv1d_update_single_token_kernel` | absent | 3933 / 17.3 ms |
| `fused_rearrange_sigmoid_gated_delta_rule_update_kernel` | absent | 3933 / 20.7 ms |
| **GDN decode window** | **73.7 ms** | **38.0 ms** |

Current main: 
image

This PR:
image

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

---

Co-authored-by: Cursor

### CODE REVIEW

**Comment by jhartika-amd** [comment]
Adding E2E perf test to this would be beneficial

**Comment by mjkvaak-amd** [comment]
> Adding E2E perf test to this would be beneficial

Perf and accuracy A/B test results added in the description.

File: .buildkite/test-amd.yaml [hunk 1]
```diff
@@ -1633,6 +1633,7 @@ steps:
   source_file_dependencies:
   - csrc/mamba/
   - tests/kernels/mamba
+  - vllm/model_executor/layers/mamba/gdn
```
Comment (dllehr-amd) [inline .buildkite/test-amd.yaml:1636]: @AndreasKaratzas Are we okay if we add this testgroup to cover the gdn change?

Comment (AndreasKaratzas) [reply .buildkite/test-amd.yaml:1837]: Yep, I think it's fine, it's not adding really a test group, it's instructing the execution of an existing test group if future PRs touch the added path, so I'm good with that

File: tests/kernels/mamba/test_gdn_rocm_layout_dispatch.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+import pytest
+import torch
+
+from vllm.model_executor.layers.mamba.gdn import qwen_gdn_linear_attn
+from vllm.model_executor.layers.mamba.gdn.qwen_gdn_linear_attn import (
+    QwenGatedDeltaNetAttention,
+)
+from vllm.v1.attention.backends.gdn_attn import GDNAttentionMetadata
+
```
Comment (tjtanaa) [inline tests/kernels/mamba/test_gdn_rocm_layout_dispatch.py:28]: @mjkvaak-amd since this is a rocm exclusive test, please guard it with pyskip so that we skip the test if it is not on rocm

Comment (mjkvaak-amd) [reply tests/kernels/mamba/test_gdn_rocm_layout_dispatch.py:28]: Thanks! Now gated behind `current_platform.is_rocm()`

**Review by tjtanaa — APPROVED** [review]
LGTM
